WPMedia/fusion-news-theme-blocks
# PEN-1395: Incorporate Google Ad work from TVNZ into an Arc Block

## Description
Added `@wpmedia/ads-block` to repo and made improvements to original code migrated from PS buildout

## Jira Ticket
- [PEN-1395](https://arcpublishing.atlassian.net/browse/PEN-1395)

## Acceptance Criteria
1. The work that was done in  in TVNZ’s repo is turned into a new Arc Block, called Google Ads – Arc Block 
2. Tests and documentation are written to cover this new functionality 

## Test Steps
Checkout branch, refresh local blocks repo (`npx lerna clean`, `npx lerna bootstrap`, etc.). 
Add `@wpmedia/ads-block` into the test Fusion repo `blocks.json`. Also add a `websiteAdPath` property of `news` into `blocks.json` in the `values.default` object. Also set `useLocal: true` in `blocks.json` Startup Fusion repo using `npx fusion start -l`.

Add one of each of the following ad units (feature is called `Google Ad - Arc Block`) to **Homepage** (page), **article-right-rail** (template) and **section** (template). Stage & publish each page after adding all ad units. Each page should have the following set of ad units placed on it:

FULL-WIDTH SECTION:
- Leaderboard - Large (set `display` to `desktop`)

MAIN SECTION:
- Leaderboard - Medium (set `display` to `desktop`)
- Cube (set `display` to `mobile`)
- Flex Cube (set `display` to `all`) - on `article-right-rail` page, add this inside `Article Body` chain
- OOP (set `display` to `all`)

RIGHT-RAIL SECTION:
- Right Rail Cube (set `display` to `all`)

Open each of these pages:
- http://localhost/pf/homepage/?_website=the-sun
- http://localhost/pf/news/2020/04/15/7-questions-about-traveling-to-australia-during-catastrophic-fires-answered/?_website=the-sun
- http://localhost/pf/entertainment/?_website=the-sun

Ensure that ads load on all of them. On each page, use the `googletag.openConsole()` command in Chrome DevTools to open the GTM dashboard to inspect each ad unit.

Check each page on both desktop and mobile breakpoints. Ensure that the ads are hidden on the proper breakpoints (i.e. ads set to `desktop` **display** should be hidden on mobile and ads set to `mobile` **display** should be hidden on desktop).

For each page, go to `Network` tab in Chrome DevTools and filter by `dfp`. Ensure that the correct amount of requests are being sent. If an ad is hidden on the current breakpoint, no request should be sent. On the desktop breakpoint, 5 requests should be sent out with the above configuration. On mobile breakpoint, only 3 requests should be sent out.

## Effect Of Changes
### Before
There was previously no working `Google Ad - Arc Block` feature in the repo

### After
There is now a working `Google Ad - Arc Block` feature available in the repo

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Add `@wpmedia/ads-block` to `blocks.json` inside `blocks` array
- Add site property of `websiteAdPath` (set to `news`) inside `blocks.json` - `values.default` object

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
